File: src/C#/BehavioralPatterns/Mediator/MultipleExamples/CommunicationHub/Participants/AccelerometerSensor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MultipleExamples.CommunicationHub.Participants
{
    public class AccelerometerSensor : Sensor
    {
        public AccelerometerSensor() : base("acceleration")
        {

        }

        public override void Receive(Participant sender, object args)
        {
            if (args == "measure")
            {
                LastValue = new Random().NextDouble();
                Mediator.Notify(this, "measure");
                Mediator.Notify(this, LastValue);
            }
        }

        public override void ValueChanged(object value)
        {
            LastValue = value;
            Mediator.Notify(this, "measure");
            Mediator.Notify(this, LastValue);
        }
    }
}

File: src/PHP/BehavioralPatterns/Mediator/CommunicationHub/Participants/AccelerometerSensor.php
<?php

namespace BehavioralPatterns\Mediator\CommunicationHub\Participants;

class AccelerometerSensor extends Sensor
{
    public function __construct()
    {
        parent::__construct("acceleration");
    }

    public function receive($sender, $args): void
    {
        if ($args === "measure")
        {
            $this->lastValue = mt_rand() / mt_getrandmax();
            $this->mediator->notify($this, "measure");
            $this->mediator->notify($this, $this->lastValue);
        }
    }

    public function valueChanged($value): void
    {
        $this->lastValue = $value;
        $this->mediator->notify($this, "measure");
        $this->mediator->notify($this, $this->lastValue);
    }
}